### PR TITLE
Add the Advanced panel to the link UI

### PR DESCRIPTION
**Pull request by filipsobol**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Feature (link): Add the Advanced panel with link decorators to the link UI. See #17230.

### CODE REVIEW

File: packages/ckeditor5-link/theme/linkform.css [hunk 1]
```diff
@@ -6,6 +6,8 @@
 @import "@ckeditor/ckeditor5-ui/theme/mixins/_rwd.css";
 
 .ck.ck-link__panel {
+	width: 340px; /* TODO: 340px instead of 300px, because of generic min-width on inputs */
```
Comment (filipsobol) [inline packages/ckeditor5-link/theme/linkform.css:9]: @dwandeljanicka

File: packages/ckeditor5-link/src/ui/linkadvancedview.ts [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public override render(): void {
+		super.render();
+
+		const childViews = [
+			...this.listChildren,
```
Comment (niegowski) [inline packages/ckeditor5-link/src/ui/linkadvancedview.ts:134]: This won't add items dynamically added to the ManualDecorator-s collection.

File: packages/ckeditor5-link/tests/linkui.js [hunk 3]
```diff
@@ -1950,37 +2044,229 @@ describe( 'LinkUI', () => {
 				it( 'should gather information about manual decorators', () => {
 					const executeSpy = testUtils.sinon.spy( editor, 'execute' );
 
-					setModelData( model, 'f[<$text linkHref="url" linkIsFoo="true">ooba</$text>]r' );
+					setModelData( model, 'f[<$text linkHref="url" foo="bar">ooba</$text>]r' );
```
Comment (niegowski) [inline packages/ckeditor5-link/tests/linkui.js:2047]: I'm not sure if this should not be adjusted so we could notice that some manual decorator is enabled.

File: packages/ckeditor5-link/tests/ui/linkadvancedview.js [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+
+			view.render();
+
+			sinon.assert.calledOnce( spy );
+			sinon.assert.calledWithExactly( spy, view.element );
+
+			view.destroy();
+		} );
+
+		describe( 'activates keyboard navigation for the toolbar', () => {
```
Comment (niegowski) [inline packages/ckeditor5-link/tests/ui/linkadvancedview.js:173]:
```suggestion
		describe( 'activates keyboard navigation', () => {
```

**Review by niegowski — APPROVED** [review]

**Pull request merged**